src/contract/ContractFunctionResult.ts

```diff
@@ -1,23 +1,81 @@
 import { ContractLogInfo, contractLogInfoListToSdk } from "./ContractLogInfo";
 import { ContractFunctionResult as ProtoContractFunctionResult } from "../generated/ContractCallLocal_pb";
 import { ContractId, contractIdToSdk } from "./ContractId";
+import BigNumber from "bignumber.js";
 
-export type ContractFunctionResult = {
-    contractId: ContractId;
-    contractCallResult: Uint8Array | string;
-    errorMessage: string;
-    bloom: Uint8Array;
-    gasUsed: number;
-    logInfoList: ContractLogInfo[];
+export class ContractFunctionResult {
+    public readonly contractId: ContractId;
+    public readonly contractCallResult: Uint8Array;
+    public readonly errorMessage: string;
+    public readonly bloom: Uint8Array;
+    public readonly gasUsed: number;
+    public readonly logInfoList: ContractLogInfo[];
+
+    // Constructor shouldn't be used since all the fields are public use that instead
```

Comment:
This comment is kind of confusing. Just say the constructor isn't part of the stable API.